# Analyze CIA Factbook Data

This project is to analyze data within the CIA Factbook. 

Dataset: https://www.cia.gov/library/publications/the-world-factbook/

Some pieces of data gathered for each country are population, land & water area, and population growth. 

## Index 

- SQL Setup & Exploring the *facts* dataset
- Summary Statistics for the *facts* dataset
- Finding Densely Populated Countries
- Population and Growth Rate
- Exploring Water-to-Land Ratio
- Comparing Death Rates and Birth Rates
- Migration Rate

## SQL Setup & Exploring the *facts* dataset

In [55]:
%%capture
%load_ext sql
%sql sqlite:///factbook.db

'Connected: None@factbook.db'

We need to see the tables in the database and the column names. 

In [56]:
%%sql
SELECT *
    FROM sqlite_master
    WHERE type = 'table';

Done.


[('table', 'sqlite_sequence', 'sqlite_sequence', 3, 'CREATE TABLE sqlite_sequence(name,seq)'),
 ('table', 'facts', 'facts', 47, 'CREATE TABLE "facts" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "code" varchar(255) NOT NULL, "name" varchar(255) NOT NULL, "area" integer, "a ... (4 characters truncated) ... land" integer, "area_water" integer, "population" integer, "population_growth" float, "birth_rate" float, "death_rate" float, "migration_rate" float)')]

The *facts* table will the table I will be conducting analysis on. The following are the first 5 rows of the *facts* table.

In [57]:
%%sql
SELECT *
    FROM facts
    LIMIT 5;

Done.


[(1, 'af', 'Afghanistan', 652230, 652230, 0, 32564342, 2.32, 38.57, 13.89, 1.51),
 (2, 'al', 'Albania', 28748, 27398, 1350, 3029278, 0.3, 12.92, 6.58, 3.3),
 (3, 'ag', 'Algeria', 2381741, 2381741, 0, 39542166, 1.84, 23.67, 4.31, 0.92),
 (4, 'an', 'Andorra', 468, 468, 0, 85580, 0.12, 8.13, 6.96, 0.0),
 (5, 'ao', 'Angola', 1246700, 1246700, 0, 19625353, 2.78, 38.78, 11.49, 0.46)]

## Summary Statistics for the *facts* table

The summary statistics are the smallest population, largest population, smallest population growth rate, and the largest population growth rate.


In [58]:
%%sql
SELECT MIN(population), MAX(population), MIN(population_growth), MAX(population_growth)
    FROM facts;


Done.


[(0, 7256490011, 0.0, 4.02)]

After pulling some summary statistics for the Population and the Population Growth columns, there are outliers. The minimum value (0) and the maximum value (7 billion+) of the population column are outliers. The next step is to analyze the countries that have the minimum or the maximum value.


In [59]:
%%sql
SELECT name
    FROM facts
    WHERE population = (SELECT MIN(population)
                           FROM facts);

Done.


[('Antarctica',)]

In [60]:
%%sql
SELECT name
    FROM facts
    WHERE population = (SELECT MAX(population)
                           FROM facts);

Done.


[('World',)]

After analyzing the outliers, the minimum value is Antartica which is a continent. The maximum value is the population of the entire world. These are not accurate reflections of what countries have the lowest and highest populations. As a result, the next step is to omit *World* and *Antarctica* from the summary statistics.

In [61]:
%%sql
SELECT MIN(population), MAX(population), MIN(population_growth), MAX(population_growth)
    FROM facts
    WHERE population != (SELECT MAX(population)
                            FROM facts);

Done.


[(0, 1367485388, 0.0, 4.02)]

After modifying the summary statistics to eliminate the 'World' population data point, the max population is around 1.3 billion. 

Next step is to pull the average population and the average land and water area while omitting the 'World' row.

In [62]:
%%sql
SELECT AVG(population) avg_population, AVG(area) avg_land_area
    FROM facts
    WHERE population != (SELECT MAX(population)
                            FROM facts);

Done.


[(32242666.56846473, 582949.8523206752)]

The average population is around 32 million and the average land and water area of the countries is around 5 million. 

Buliding upon the average population and average land & water area results, the next query will help see the countries where the population is above average AND the area is below average. 

## Finding Densely Populated Countries

In [63]:
%%sql
SELECT *
    FROM facts
    WHERE population > (SELECT AVG(population)
                           FROM facts
                           WHERE name <> 'World') 
        AND area < (SELECT AVG(area)
                        FROM facts
                        WHERE name <> 'World');

Done.


[(14, 'bg', 'Bangladesh', 148460, 130170, 18290, 168957745, 1.6, 21.14, 5.61, 0.46),
 (65, 'gm', 'Germany', 357022, 348672, 8350, 80854408, 0.17, 8.47, 11.42, 1.24),
 (80, 'iz', 'Iraq', 438317, 437367, 950, 37056169, 2.93, 31.45, 3.77, 1.62),
 (83, 'it', 'Italy', 301340, 294140, 7200, 61855120, 0.27, 8.74, 10.19, 4.1),
 (85, 'ja', 'Japan', 377915, 364485, 13430, 126919659, 0.16, 7.93, 9.51, 0.0),
 (91, 'ks', 'Korea, South', 99720, 96920, 2800, 49115196, 0.14, 8.19, 6.75, 0.0),
 (120, 'mo', 'Morocco', 446550, 446300, 250, 33322699, 1.0, 18.2, 4.81, 3.36),
 (138, 'rp', 'Philippines', 300000, 298170, 1830, 100998376, 1.61, 24.27, 6.11, 2.09),
 (139, 'pl', 'Poland', 312685, 304255, 8430, 38562189, 0.09, 9.74, 10.19, 0.46),
 (163, 'sp', 'Spain', 505370, 498980, 6390, 48146134, 0.89, 9.64, 9.04, 8.31),
 (173, 'th', 'Thailand', 513120, 510890, 2230, 67976405, 0.34, 11.19, 7.8, 0.0),
 (182, 'ug', 'Uganda', 241038, 197100, 43938, 37101745, 3.24, 43.79, 10.69, 0.74),
 (185, 'uk', 'United Kingdom', 243610, 241930, 1680, 64088222, 0.54, 12.17, 9.35, 2.54),
 (192, 'vm', 'Vietnam', 331210, 310070, 21140, 94348835, 0.97, 15.96, 5.93, 0.3)]

Some of the densely populated countries are Bangladesh, Germany, Iraq, the Phillippines, Thailand, United Kingdom and Vietnam. 

## Population and Growth Rate

This step will examine the most populous country and the country with the highest growth rate.

In [64]:
%%sql
SELECT * 
    FROM facts
    WHERE name <> 'World'
    ORDER BY population DESC
    LIMIT 1

Done.


[(37, 'ch', 'China', 9596960, 9326410, 270550, 1367485388, 0.45, 12.49, 7.53, 0.44)]

In [65]:
%%sql
SELECT * 
    FROM facts
    WHERE name <> 'World'
    ORDER BY population_growth DESC
    LIMIT 1

Done.


[(162, 'od', 'South Sudan', 644329, None, None, 12042910, 4.02, 36.91, 8.18, 11.47)]

The country with the highest population is China with 1.3 billion people; however, the country with the highest population growth rate is South Sudan with a rate of 4.02.

## Exploring Water-to-Land Ratio

This section pulls the countries with the highest ratios of water to land, and the countries that have more water than land. 

In [66]:
%%sql
SELECT name,
       area_land,
       area_water,
       CAST(area_water AS Float) / CAST(area_land AS Float) AS water_land_ratio
    FROM facts
    WHERE water_land_ratio > 0.0
    ORDER BY water_land_ratio DESC

Done.


[('British Indian Ocean Territory', 60, 54340, 905.6666666666666),
 ('Virgin Islands', 346, 1564, 4.520231213872832),
 ('Puerto Rico', 8870, 4921, 0.5547914317925592),
 ('Bahamas, The', 10010, 3870, 0.3866133866133866),
 ('Guinea-Bissau', 28120, 8005, 0.2846728307254623),
 ('Malawi', 94080, 24404, 0.25939625850340137),
 ('Netherlands', 33893, 7650, 0.22571032366565366),
 ('Uganda', 197100, 43938, 0.22292237442922375),
 ('Eritrea', 101000, 16600, 0.16435643564356436),
 ('Liberia', 96320, 15049, 0.15623961794019933),
 ('Bangladesh', 130170, 18290, 0.1405085657217485),
 ('Gambia, The', 10120, 1180, 0.116600790513834),
 ('Taiwan', 32260, 3720, 0.1153130812151271),
 ('Finland', 303815, 34330, 0.11299639583299047),
 ('India', 2973193, 314070, 0.10563390940312317),
 ('Canada', 9093507, 891163, 0.09799992456155804),
 ('Sweden', 410335, 39960, 0.09738384490720996),
 ('Colombia', 1038700, 100210, 0.09647636468662751),
 ('Brunei', 5265, 500, 0.0949667616334283),
 ('Guyana', 196849, 18120, 0.09205025171578214),
 ('French Polynesia', 3827, 340, 0.08884243532793311),
 ('Nicaragua', 119990, 10380, 0.08650720893407784),
 ('Burundi', 25680, 2150, 0.0837227414330218),
 ('Iran', 1531595, 116600, 0.07612978626856316),
 ('United States', 9161966, 664709, 0.07255091319919764),
 ('Tanzania', 885800, 61500, 0.06942876495822985),
 ('Vietnam', 310070, 21140, 0.06817815332021801),
 ('Rwanda', 24668, 1670, 0.06769904329495703),
 ('Estonia', 42388, 2840, 0.06700009436633009),
 ('Norway', 304282, 19520, 0.06415101780585115),
 ('Armenia', 28203, 1540, 0.054604120129064286),
 ('Tunisia', 155360, 8250, 0.05310247167868177),
 ('Uzbekistan', 425400, 22000, 0.051716031969910675),
 ('Indonesia', 1811569, 93000, 0.0513367141963679),
 ('Suriname', 156000, 7820, 0.05012820512820513),
 ('Albania', 27398, 1350, 0.04927366961092051),
 ('Ghana', 227533, 11000, 0.048344635723169826),
 ('Azerbaijan', 82629, 3971, 0.048058187803313605),
 ('Togo', 54385, 2400, 0.044129815206398824),
 ('Russia', 16377742, 720500, 0.04399263341674328),
 ('Kyrgyzstan', 191801, 8150, 0.04249195781043894),
 ('Tonga', 717, 30, 0.04184100418410042),
 ('Ukraine', 579330, 24220, 0.04180691488443547),
 ('Lithuania', 62680, 2620, 0.041799617102744095),
 ('West Bank', 5640, 220, 0.03900709219858156),
 ('Gabon', 257667, 10000, 0.03880978161735884),
 ('Turkmenistan', 469930, 18170, 0.038665333134722195),
 ('Hungary', 89608, 3420, 0.038166235157575217),
 ('Latvia', 62249, 2340, 0.03759096531671192),
 ('Romania', 229891, 8500, 0.03697404422095689),
 ('Japan', 364485, 13430, 0.036846509458551106),
 ('Burma', 653508, 23070, 0.03530178666519767),
 ('Congo, Democratic Republic of the', 2267048, 77810, 0.03432216697661452),
 ('Venezuela', 882050, 30000, 0.034011677342554274),
 ('Pakistan', 770875, 25220, 0.03271606940165397),
 ('Hong Kong', 1073, 35, 0.032618825722273995),
 ('Solomon Islands', 27986, 910, 0.03251625812906453),
 ('Switzerland', 39997, 1280, 0.032002400180013504),
 ('Moldova', 32891, 960, 0.029187315679061143),
 ('China', 9326410, 270550, 0.029009018475490568),
 ('Korea, South', 96920, 2800, 0.028889806025588115),
 ('Poland', 304255, 8430, 0.027707022070302872),
 ('Iceland', 100250, 2750, 0.02743142144638404),
 ('Montenegro', 13452, 360, 0.026761819803746655),
 ('Nepal', 143351, 3830, 0.02671763712844696),
 ('Botswana', 566730, 15000, 0.026467630088401885),
 ('Laos', 230800, 6000, 0.025996533795493933),
 ('Cambodia', 176515, 4520, 0.025606888932951873),
 ('Italy', 294140, 7200, 0.024478139661385734),
 ('Ecuador', 276841, 6720, 0.02427386116940771),
 ('Germany', 348672, 8350, 0.023948008443465493),
 ('Paraguay', 397302, 9450, 0.023785432743857318),
 ('Belarus', 202900, 4700, 0.023164120256283883),
 ('Papua New Guinea', 452860, 9980, 0.022037715850373184),
 ('Bulgaria', 108489, 2390, 0.022029883213966393),
 ('Senegal', 192530, 4192, 0.02177323014595128),
 ('Israel', 20330, 440, 0.02164289227742253),
 ('Czech Republic', 77247, 1620, 0.02097168822090178),
 ('Ireland', 68883, 1390, 0.02017914434621024

The countries with the highest water to land ratios are the British Indian Ocean Territory, the Virgin Islands, Puerto Rico, The Bahamas and Guinea-Bissau. British Indian Ocean Territory and Virgin Islands have more water than land. 

## Comparing Death Rates and Birth Rates

Which countries have a higher death rate than birth rate and vice versa?

In [67]:
%%sql
SELECT name,
       death_rate,
       birth_rate
FROM facts

Done.


[('Afghanistan', 13.89, 38.57),
 ('Albania', 6.58, 12.92),
 ('Algeria', 4.31, 23.67),
 ('Andorra', 6.96, 8.13),
 ('Angola', 11.49, 38.78),
 ('Antigua and Barbuda', 5.69, 15.85),
 ('Argentina', 7.33, 16.64),
 ('Armenia', 9.34, 13.61),
 ('Australia', 7.14, 12.15),
 ('Austria', 9.42, 9.41),
 ('Azerbaijan', 7.07, 16.64),
 ('Bahamas, The', 7.05, 15.5),
 ('Bahrain', 2.69, 13.66),
 ('Bangladesh', 5.61, 21.14),
 ('Barbados', 8.44, 11.87),
 ('Belarus', 13.36, 10.7),
 ('Belgium', 9.63, 11.41),
 ('Belize', 5.97, 24.68),
 ('Benin', 8.21, 36.02),
 ('Bhutan', 6.69, 17.78),
 ('Bolivia', 6.52, 22.76),
 ('Bosnia and Herzegovina', 9.75, 8.87),
 ('Botswana', 13.39, 20.96),
 ('Brazil', 6.58, 14.46),
 ('Brunei', 3.52, 17.32),
 ('Bulgaria', 14.44, 8.92),
 ('Burkina Faso', 11.72, 42.03),
 ('Burma', 7.96, 18.39),
 ('Burundi', 9.27, 42.01),
 ('Cambodia', 7.68, 23.83),
 ('Cameroon', 10.11, 36.17),
 ('Canada', 8.42, 10.28),
 ('Cabo Verde', 6.11, 20.33),
 ('Central African Republic', 13.8, 35.08),
 ('Chad', 14.28, 36.6),
 ('Chile', 6.0, 13.83),
 ('China', 7.53, 12.49),
 ('Colombia', 5.4, 16.47),
 ('Comoros', 7.57, 27.84),
 ('Congo, Democratic Republic of the', 10.07, 34.88),
 ('Congo, Republic of the', 10.0, 35.85),
 ('Costa Rica', 4.55, 15.91),
 ("Cote d'Ivoire", 9.55, 28.67),
 ('Croatia', 12.18, 9.45),
 ('Cuba', 7.72, 9.9),
 ('Cyprus', 6.62, 11.41),
 ('Czech Republic', 10.34, 9.63),
 ('Denmark', 10.25, 10.27),
 ('Djibouti', 7.73, 23.65),
 ('Dominica', 7.91, 15.41),
 ('Dominican Republic', 4.55, 18.73),
 ('Ecuador', 5.06, 18.51),
 ('Egypt', 4.77, 22.9),
 ('El Salvador', 5.69, 16.46),
 ('Equatorial Guinea', 8.19, 33.31),
 ('Eritrea', 7.52, 30.0),
 ('Estonia', 12.4, 10.51),
 ('Ethiopia', 8.19, 37.27),
 ('Fiji', 6.04, 19.43),
 ('Finland', 9.83, 10.72),
 ('France', 9.16, 12.38),
 ('Gabon', 13.12, 34.49),
 ('Gambia, The', 7.15, 30.86),
 ('Georgia', 10.82, 12.74),
 ('Germany', 11.42, 8.47),
 ('Ghana', 7.22, 31.09),
 ('Greece', 11.09, 8.66),
 ('Grenada', 8.08, 16.03),
 ('Guatemala', 4.77, 24.89),
 ('Guinea', 9.46, 35.74),
 ('Guinea-Bissau', 14.33, 33.38),
 ('Guyana', 7.32, 15.59),
 ('Haiti', 7.83, 22.31),
 ('Honduras', 5.17, 23.14),
 ('Hungary', 12.73, 9.16),
 ('Iceland', 6.28, 13.91),
 ('India', 7.32, 19.55),
 ('Indonesia', 6.37, 16.72),
 ('Iran', 5.94, 17.99),
 ('Iraq', 3.77, 31.45),
 ('Ireland', 6.48, 14.84),
 ('Israel', 5.15, 18.48),
 ('Italy', 10.19, 8.74),
 ('Jamaica', 6.7, 18.16),
 ('Japan', 9.51, 7.93),
 ('Jordan', 3.79, 25.37),
 ('Kazakhstan', 8.21, 19.15),
 ('Kenya', 6.89, 26.4),
 ('Kiribati', 7.12, 21.46),
 ('Korea, North', 9.21, 14.52),
 ('Korea, South', 6.75, 8.19),
 ('Kosovo', None, None),
 ('Kuwait', 2.18, 19.91),
 ('Kyrgyzstan', 6.65, 22.98),
 ('Laos', 7.63, 24.25),
 ('Latvia', 14.31, 10.0),
 ('Lebanon', 4.88, 14.59),
 ('Lesotho', 14.89, 25.47),
 ('Liberia', 9.69, 34.41),
 ('Libya', 3.58, 18.03),
 ('Liechtenstein', 7.12, 10.45),
 ('Lithuania', 14.27, 10.1),
 ('Luxembourg', 7.24, 11.37),
 ('Macedonia', 9.08, 11.55),
 ('Madagascar', 6.81, 32.61),
 ('Malawi', 8.41, 41.56),
 ('Malaysia', 5.03, 19.71),
 ('Maldives', 3.89, 15.75),
 ('Mali', 12.89, 44.99),
 ('Malta', 9.09, 10.18),
 ('Marshall Islands', 4.21, 25.6),
 ('Mauritania', 8.2, 31.34),
 ('Mauritius', 6.91, 13.29),
 ('Mexico', 5.26, 18.78),
 ('Micronesia, Federated States of', 4.23, 20.54),
 ('Moldova', 12.59, 12.0),
 ('Monaco', 9.24, 6.65),
 ('Mongolia', 6.35, 20.25),
 ('Montenegro', 9.43, 10.42),
 ('Morocco', 4.81, 18.2),
 ('Mozambique', 12.1, 38.58),
 ('Namibia', 13.91, 19.8),
 ('Nauru', 5.87, 24.95),
 ('Nepal', 6.56, 20.64),
 ('Netherlands', 8.66, 10.83),
 ('New Zealand', 7.36, 13.33),
 ('Nicaragua', 5.08, 18.03),
 ('Niger', 12.42, 45.45),
 ('Nigeria', 12.9, 37.64),
 ('Norway', 8.12, 12.14),
 ('Oman', 3.36, 24.44),
 ('Pakistan', 6.49, 22.58),
 ('Palau', 7.99, 11.05),
 ('Panama', 4.81, 18.32),
 ('Papua New Guinea', 6.53, 24.38),
 ('Paraguay', 4.68, 16.37),
 ('Peru', 6.01, 18.28),
 ('Philippines', 6.11, 24.27),
 ('Poland', 10.19, 9.74),
 ('Portugal', 11.02, 9.27),
 ('Qatar', 1.53, 9.84),
 ('R

In the birth_rate and death_rate, there are 'None' values, so these have to be taken into account in our queries comparing the values.

With this next query, I seek to determine which countries have a higher death rate than birth rate. 

In [68]:
%%sql
SELECT name,
       death_rate,
       birth_rate,
       CAST(death_rate AS Float) - CAST(birth_rate AS Float) death_birth_diff
FROM facts
WHERE birth_rate != 'None'
    AND death_rate != 'None'
    AND death_rate > birth_rate
ORDER BY death_birth_diff DESC

Done.


[('Bulgaria', 14.44, 8.92, 5.52),
 ('Serbia', 13.66, 9.08, 4.58),
 ('Latvia', 14.31, 10.0, 4.3100000000000005),
 ('Lithuania', 14.27, 10.1, 4.17),
 ('Ukraine', 14.46, 10.72, 3.74),
 ('Hungary', 12.73, 9.16, 3.5700000000000003),
 ('Germany', 11.42, 8.47, 2.9499999999999993),
 ('Slovenia', 11.37, 8.42, 2.9499999999999993),
 ('Romania', 11.9, 9.14, 2.76),
 ('Croatia', 12.18, 9.45, 2.7300000000000004),
 ('Belarus', 13.36, 10.7, 2.66),
 ('Monaco', 9.24, 6.65, 2.59),
 ('Greece', 11.09, 8.66, 2.4299999999999997),
 ('Saint Pierre and Miquelon', 9.72, 7.42, 2.3000000000000007),
 ('Russia', 13.69, 11.6, 2.09),
 ('Estonia', 12.4, 10.51, 1.8900000000000006),
 ('Portugal', 11.02, 9.27, 1.75),
 ('Japan', 9.51, 7.93, 1.58),
 ('Italy', 10.19, 8.74, 1.4499999999999993),
 ('Bosnia and Herzegovina', 9.75, 8.87, 0.8800000000000008),
 ('Czech Republic', 10.34, 9.63, 0.7099999999999991),
 ('Moldova', 12.59, 12.0, 0.5899999999999999),
 ('Poland', 10.19, 9.74, 0.4499999999999993),
 ('Austria', 9.42, 9.41, 0.009999999999999787)]

There are 24 countries in the factbook that have death rates higher than birth rates. The top 5 countries with the largest difference between their death rates and their birth rates are *Bulgaria, Serbia, Latvia, Lithuania, and Ukraine.* Furthermore, most of the countries with higher death rates than birth rates are located in Europe. 

Next is to see which countries have a birth rate higher than a death rate. 

In [69]:
%%sql
SELECT name, 
       birth_rate,
       death_rate,
       CAST(birth_rate AS Float) - CAST(death_rate AS Float) birth_death_diff
FROM facts
WHERE birth_rate != 'None'
    AND death_rate != 'None'
    AND birth_rate > death_rate
ORDER BY birth_death_diff DESC
       

Done.


[('Malawi', 41.56, 8.41, 33.150000000000006),
 ('Uganda', 43.79, 10.69, 33.1),
 ('Niger', 45.45, 12.42, 33.03),
 ('Burundi', 42.01, 9.27, 32.739999999999995),
 ('Mali', 44.99, 12.89, 32.1),
 ('Burkina Faso', 42.03, 11.72, 30.310000000000002),
 ('Zambia', 42.13, 12.67, 29.46),
 ('Ethiopia', 37.27, 8.19, 29.080000000000005),
 ('South Sudan', 36.91, 8.18, 28.729999999999997),
 ('Tanzania', 36.39, 8.0, 28.39),
 ('Gaza Strip', 31.11, 3.04, 28.07),
 ('Timor-Leste', 34.16, 6.1, 28.059999999999995),
 ('Benin', 36.02, 8.21, 27.810000000000002),
 ('Iraq', 31.45, 3.77, 27.68),
 ('Angola', 38.78, 11.49, 27.29),
 ('Sao Tome and Principe', 34.23, 7.24, 26.989999999999995),
 ('Togo', 34.13, 7.26, 26.870000000000005),
 ('Somalia', 40.45, 13.62, 26.830000000000005),
 ('Mozambique', 38.58, 12.1, 26.479999999999997),
 ('Guinea', 35.74, 9.46, 26.28),
 ('Sierra Leone', 37.03, 10.81, 26.22),
 ('Cameroon', 36.17, 10.11, 26.060000000000002),
 ('Senegal', 34.52, 8.46, 26.060000000000002),
 ('Congo, Republic of the', 35.85, 10.0, 25.85),
 ('Madagascar', 32.61, 6.81, 25.8),
 ('Equatorial Guinea', 33.31, 8.19, 25.120000000000005),
 ('Congo, Democratic Republic of the', 34.88, 10.07, 24.810000000000002),
 ('Rwanda', 33.75, 8.96, 24.79),
 ('Nigeria', 37.64, 12.9, 24.740000000000002),
 ('Liberia', 34.41, 9.69, 24.72),
 ('Afghanistan', 38.57, 13.89, 24.68),
 ('Ghana', 31.09, 7.22, 23.87),
 ('Gambia, The', 30.86, 7.15, 23.71),
 ('Yemen', 29.98, 6.28, 23.7),
 ('Mauritania', 31.34, 8.2, 23.14),
 ('Eritrea', 30.0, 7.52, 22.48),
 ('Chad', 36.6, 14.28, 22.32),
 ('Zimbabwe', 32.26, 10.13, 22.129999999999995),
 ('Solomon Islands', 25.77, 3.85, 21.919999999999998),
 ('Western Sahara', 30.24, 8.34, 21.9),
 ('Jordan', 25.37, 3.79, 21.580000000000002),
 ('Sudan', 29.19, 7.66, 21.53),
 ('Marshall Islands', 25.6, 4.21, 21.39),
 ('Gabon', 34.49, 13.12, 21.370000000000005),
 ('Central African Republic', 35.08, 13.8, 21.279999999999998),
 ('Oman', 24.44, 3.36, 21.080000000000002),
 ('Vanuatu', 25.04, 4.09, 20.95),
 ('Comoros', 27.84, 7.57, 20.27),
 ('Guatemala', 24.89, 4.77, 20.12),
 ('Kenya', 26.4, 6.89, 19.509999999999998),
 ('West Bank', 22.99, 3.5, 19.49),
 ('Algeria', 23.67, 4.31, 19.360000000000003),
 ("Cote d'Ivoire", 28.67, 9.55, 19.12),
 ('Nauru', 24.95, 5.87, 19.08),
 ('Guinea-Bissau', 33.38, 14.33, 19.050000000000004),
 ('Belize', 24.68, 5.97, 18.71),
 ('Tajikistan', 24.38, 6.18, 18.2),
 ('Syria', 22.17, 4.0, 18.17),
 ('Philippines', 24.27, 6.11, 18.16),
 ('Tonga', 23.0, 4.85, 18.15),
 ('American Samoa', 22.89, 4.75, 18.14),
 ('Egypt', 22.9, 4.77, 18.13),
 ('Honduras', 23.14, 5.17, 17.97),
 ('Papua New Guinea', 24.38, 6.53, 17.849999999999998),
 ('Kuwait', 19.91, 2.18, 17.73),
 ('Laos', 24.25, 7.63, 16.62),
 ('Kyrgyzstan', 22.98, 6.65, 16.33),
 ('Micronesia, Federated States of', 20.54, 4.23, 16.31),
 ('Bolivia', 22.76, 6.52, 16.240000000000002),
 ('Cambodia', 23.83, 7.68, 16.15),
 ('Pakistan', 22.58, 6.49, 16.089999999999996),
 ('Djibouti', 23.65, 7.73, 15.919999999999998),
 ('Samoa', 20.87, 5.32, 15.55),
 ('Bangladesh', 21.14, 5.61, 15.530000000000001),
 ('Saudi Arabia', 18.51, 3.33, 15.180000000000001),
 ('Tuvalu', 23.74, 8.74, 14.999999999999998),
 ('Malaysia', 19.71, 5.03, 14.68),
 ('Northern Mariana Islands', 18.32, 3.71, 14.61),
 ('Haiti', 22.31, 7.83, 14.479999999999999),
 ('Libya', 18.03, 3.58, 14.450000000000001),
 ('Kiribati', 21.46, 7.12, 14.34),
 ('Cabo Verde', 20.33, 6.11, 14.219999999999999),
 ('Dominican Republic', 18.73, 4.55, 14.18),
 ('Nepal', 20.64, 6.56, 14.080000000000002),
 ('Mongolia', 20.25, 6.35, 13.9),
 ('Venezuela', 19.16, 5.31, 13.850000000000001),
 ('Brunei', 17.32, 3.52, 13.8),
 ('Mexico', 18.78, 5.26, 13.520000000000001),
 ('Panama', 18.32, 4.81, 13.510000000000002),
 ('United Arab Emirates', 15.43, 1.97, 13.459999999999999),
 ('Ecuador', 18.51, 5.06, 13.450000000000003),
 ('Fiji', 19.43, 6.04, 13.39),
 ('Morocco', 18.2, 4.81, 13.39),
 ('Israel', 18.48, 5.15, 13.33),
 ('Turkmenistan', 19.4, 6.13, 13.27),
 ('Turks and Caicos Islands'

There are far more countries that have a birth rate that is higher than their death rate. The top 5 countries with the largest differences between their birth rate and death rate are *Malawi, Uganda, Niger, Burundi, and Mali.* All located in the African continent. 

## Migration Rate

In this final section, I want to explore countries and their migration rates. Which countries have the highest or lowest migration rates?

In [70]:
%%sql
SELECT name,
       migration_rate
FROM facts
WHERE migration_rate != 'None'
ORDER BY migration_rate DESC
LIMIT 5

Done.


[('Qatar', 22.39),
 ('American Samoa', 21.13),
 ('Micronesia, Federated States of', 20.93),
 ('Syria', 19.79),
 ('Tonga', 17.84)]

In [71]:
%%sql 
SELECT name,
       migration_rate
FROM facts
WHERE migration_rate == 0.0

Done.


[('Andorra', 0.0),
 ('Argentina', 0.0),
 ('Azerbaijan', 0.0),
 ('Bahamas, The', 0.0),
 ('Belize', 0.0),
 ('Benin', 0.0),
 ('Bhutan', 0.0),
 ('Burkina Faso', 0.0),
 ('Burundi', 0.0),
 ('Central African Republic', 0.0),
 ("Cote d'Ivoire", 0.0),
 ('Ecuador', 0.0),
 ('Equatorial Guinea', 0.0),
 ('Eritrea', 0.0),
 ('Guinea', 0.0),
 ('Guinea-Bissau', 0.0),
 ('Japan', 0.0),
 ('Korea, South', 0.0),
 ('Liberia', 0.0),
 ('Madagascar', 0.0),
 ('Malawi', 0.0),
 ('Mauritius', 0.0),
 ('Namibia', 0.0),
 ('Papua New Guinea', 0.0),
 ('Serbia', 0.0),
 ('Swaziland', 0.0),
 ('Thailand', 0.0),
 ('Togo', 0.0),
 ('Venezuela', 0.0),
 ('Zimbabwe', 0.0),
 ('Faroe Islands', 0.0),
 ('Montserrat', 0.0),
 ('Saint Helena, Ascension, and Tristan da Cunha', 0.0),
 ('Gaza Strip', 0.0),
 ('West Bank', 0.0)]

The countries with the highest migration rate is Qatar at 22.39% which means that there are far more immigrants than emigrants in the country. There are several countries tied with a rate of 0.0. 